lucene/sandbox/src/java/org/apache/lucene/sandbox/search/MultiNormsLeafSimScorer.java

```diff
@@ -128,7 +128,7 @@ public boolean advanceExact(int target) throws IOException {
       for (int i = 0; i < normsArr.length; i++) {
         boolean found = normsArr[i].advanceExact(target);
         assert found;
-        normValue += weightArr[i] * LENGTH_TABLE[(byte) normsArr[i].longValue()];
+        normValue += weightArr[i] * LENGTH_TABLE[(byte) normsArr[i].longValue() & 0xff];
```

Comment:
Actually, I see that `SimilarityBase.java` uses `Byte.toUnsignedInt` instead -- should we use that here?  It is the same computation?

Or, if you want to stick with `& 0xff`, could you add parens around the `(byte)` cast to make the order of operations clear?

Reply:
I've updated the change using Byte.toUnsignedInt